src/node/snapshotter.h

```diff
@@ -191,7 +193,7 @@ namespace ccf
 
       for (auto it = pending_snapshots.begin(); it != pending_snapshots.end();)
       {
-        if (idx > it->evidence_idx)
+        if (idx > it->evidence_idx && it->sig.has_value())
```

Comment:
```suggestion
        if (idx > it->evidence_idx)
```

This can go once the hook become local commit hooks (`set_map_hook`).